src/GafferScene/RenderController.cpp

```diff
@@ -316,6 +330,35 @@ class RenderController::SceneGraph
 						m_changedComponents |= AttributesComponent;
 					}
 				}
+
+				if( ( m_changedComponents & AttributesComponent ) || ( changedGlobals & ( TransformBlurGlobalComponent | DeformationBlurGlobalComponent ) ) )
+				{
+					// If attributes have changed, need to check if this has affected our motion sampling
+					int transformSegments = motionSegments( controller->m_motionBlurOptions.transformBlur, g_transformBlurAttributeName, g_transformBlurSegmentsAttributeName );
+					int deformationSegments = motionSegments( controller->m_motionBlurOptions.deformationBlur, g_deformationBlurAttributeName, g_deformationBlurSegmentsAttributeName );
+
+					if( transformSegments != m_transformSegments )
+					{
+						m_transformSegments = transformSegments;
+						m_dirtyComponents |= TransformComponent;
+					}
+
+					if( deformationSegments != m_deformationSegments )
+					{
+						m_deformationSegments = deformationSegments;
+						m_dirtyComponents |= ObjectComponent;
+					}
+				}
+			}
+
+			if( changedGlobals & TransformBlurGlobalComponent )
+			{
+				m_dirtyComponents |= TransformComponent;
+			}
```

Comment:
I presume there is a reason why we need to do this even if we discovered above that `m_transformSegments` wasn't actually changing? It's not obvious to me though, so I think needs an explanatory comment.

Reply:
I think this was left over from an earlier version and is no longer necessary.  Removed in 6a932f6